### ISSUE TITLE
Combinando encoder e motor para controlar caminho

### ISSUE THREAD

**Issue by jcert** [HackerSpaceMaringa/FutebolRobosArduino#2]
Como o robô vai andar? O server que vai cuidar pra ele ir até o ponto final? dando velocidade e direção mas não a distância? Pois eu acho que se combinar os valores do encoder e o acionamento dos motores pra fazer uma função "ande_frente(cm)"  que no mínimo ia tirar um pouco de trabalho centralizado e dar melhor precisão pra caminhada deles

**Comment by ysiraichi** [comment]
Hmmm. Boa ideia! Podemos fazer isso, sem problemas (eu acho). @jcert você tem alguma ideia ou sugestão do cálculo? 
Eu estava pensando em pegar o raio da roda, e o rpm, e estimar os segundos em que os motores seriam ligados.
O que você acha?

**Comment by LionsWrath** [comment]
Eu estava pensando nisso. Podemos deixar que o server cuide do envio dos comandos e mensagens de parada, mas pode ser que o delay do processamento + envio da mensagem deixe as rotas incorretas (e imprecisas). Talvez passar a velocidade, direção e distância seja melhor, mas isso tem que ser testado.

**Comment by ysiraichi** [comment]
Sim. Acho que podemos atrasar essas decisões para quando o robô estiver andando. Mas, ainda continuar pensando em formas de fazer isso.

**Comment by jcert** [comment]
Dá pra ir fazendo e só incluir quando o robô estiver andando, como um upgrade. Minha sugestão é não estimar tempo, e sim fazer um loop de controle, liga o loop com a distância, e talvez a rotação, e ele usa o encoder pra andar o suficiente. Talvez dê pra fazer a parte de controle com uma interrupção, daí não impede a comunicação do Blu.
